src/main/ts/components/Editor.ts

```diff
@@ -6,102 +6,109 @@
+    const modelBind: boolean = ctx.attrs['onUpdate:modelValue'] ? true : false;
+    let mounting: boolean = true;
+    const initialValue: string = props.initialValue ? props.initialValue : '';
+    let cache: string = '';
 
-    if (getTinymce() !== null) {
-      initialise(this)();
-    } else if (this.element && this.element.ownerDocument) {
-      const channel = this.$props.cloudChannel ? this.$props.cloudChannel : '5';
-      const apiKey = this.$props.apiKey ? this.$props.apiKey : 'no-api-key';
+    const getContent = (): string => {
+      return modelBind ? modelValue.value : mounting ? initialValue : cache;
+    };
 
-      const scriptSrc = isNullOrUndefined(this.$props.tinymceScriptSrc) ?
-        `https://cdn.tiny.cloud/1/${apiKey}/tinymce/${channel}/tinymce.min.js` :
-        this.$props.tinymceScriptSrc;
-
-      ScriptLoader.load(
-        this.element.ownerDocument,
-        scriptSrc,
-        initialise(this)
-      );
+    const initWrapper = (): void => {
+      const finalInit = {
+        ...props.init,
+        readonly: props.disabled,
+        selector: `#${elementId}`,
+        plugins: mergePlugins(props.init && props.init.plugins, props.plugins),
+        toolbar: props.toolbar || (props.init && props.init.toolbar),
+        inline: inlineEditor,
+        setup: (editor: any) => {
+          vueEditor = editor;
+          editor.on('init', (e: Event) => initEditor(e, props, ctx, editor, modelValue, getContent()));
+          if (props.init && typeof props.init.setup === 'function') {
+            props.init.setup(editor);
+          }
+        }
+      };
+      if (isTextarea(element.value)) {
+        element.value.style.visibility = '';
+      }
+      getTinymce().init(finalInit);
+      mounting = false;
+    };
+    watch(disabled, (disable) => {
+      if (vueEditor !== null) {
+        vueEditor.setMode(disable ? 'readonly' : 'design');
+      }
+    });
+    onMounted(() => {
+      if (getTinymce() !== null) {
+        initWrapper();
+      } else if (element.value && element.value.ownerDocument) {
+        const channel = props.cloudChannel ? props.cloudChannel : '5';
+        const apiKey = props.apiKey ? props.apiKey : 'no-api-key';
+        const scriptSrc = isNullOrUndefined(props.tinymceScriptSrc) ?
+          `https://cdn.tiny.cloud/1/${apiKey}/tinymce/${channel}/tinymce.min.js` :
+          props.tinymceScriptSrc;
+        ScriptLoader.load(
+          element.value.ownerDocument,
+          scriptSrc,
+          initWrapper
+        );
+      }
+    });
+    onBeforeUnmount(() => {
+      if (getTinymce() !== null) {
+        getTinymce().remove(vueEditor);
+      }
+    });
+    if (!inlineEditor) {
+      onActivated(() => {
+        if (!mounting) {
+          initWrapper();
+        }
+      });
+      onDeactivated(() => {
+        if (!modelBind) {
+          cache = vueEditor.getContent({ format: props.outputFormat });
```

Comment:
This is for retaining the content between initializations of the editor when using keep-live, right? In that case, do we want the format to always be 'html' here? as otherwise you could lose some of the content in the teardown->setup process.

Can we test this? I know that tinymce-vue has fallen short when it comes to tests so any test is a good test

Reply:
Mmm I agree. I wanted to keep a consistent behaviour between using `outputformat:'text'` between keep-alive and modelValue however, the real issue I have to address is the outputFormat on the binding.

Reply:
Nvm this, I was thinking of something else. This has been addressed.